Notes/NotesListView.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  NotesListView.swift
+//  Notes
+//
+//  Created by Lena Vadakkel on 09.11.23.
+//
+
+import SwiftUI
+
+struct NotesListView: View {
+    @StateObject var viewModel = NotesListViewModel()
+    @FetchRequest var notesList: FetchedResults<Note>
+    let username: String
+    
+    init(username: String) {
```

Comment:
ViewModel should be the source of truth of this View. `username` should be stored in the ViewModel
When you declare the viewModel as `var`, it will create for you the init by default

```
struct NotesListView: View {
     @StateObject var viewModel: NotesListViewModel

     var body: some View { ...}
}
```

then, username and notesList need to be moved into ViewModel. Like that you can test everything.

Reply:
username have to stay there because It gets the value from an other view and the notesList also have to be there because when I tried it to do the fetch in the ViewModel and give it over to the view, it wouldn't update if I delete or add an note the view .

Reply:
We can create the ViewModel from the the username then pass it when we create the View

the code will look like that:

```
struct OtherView: View {
    var body: some View {
       ...
       NavigationLink(
           destination: NotesListView(
              viewModel: NotesListViewModel(username: someUserName)
            ),
           title: ...
       )
    }
}
```

Reply:
I moved this to the VM but when the list is created it shows none of the notes, how can I solve this?